pageserver/src/pgdatadir_mapping.rs

```diff
@@ -2092,7 +2092,11 @@ impl DatadirModification<'_> {
                 self.tline.aux_file_size_estimator.on_add(content.len());
                 new_files.push((path, content));
             }
-            (None, true) => warn!("removing non-existing aux file: {}", path),
+            // Compute may request delete of old version of pgstat AUX file if new one exceeds size limit.
+            // Compute doesn't know if previous version of this file exists or not, so
+            // attempt to delete non-existing file can cause this message.
+            // To avoid false alarms, log it as info rather than warning.
+            (None, true) => info!("removing non-existing aux file: {}", path),
```

Comment:
Could we do this to make the exception more narrow?
```suggestion
            // Compute may request delete of old version of pgstat AUX file if new one exceeds size limit.
            // Compute doesn't know if previous version of this file exists or not, so
            // attempt to delete non-existing file can cause this message.
            // To avoid false alarms, log it as info rather than warning.
            (None, true) if path.starts_with("pg_stat/") => info!("removing non-existing pg_stat file: {}", path),
            (None, true)  => warn!("removing non-existing aux file: {}", path),
```